1-Draft/RFC00XX-Update-Error-View.md

```diff
@@ -0,0 +1,163 @@
+new or occasional PowerShell users.
+To improve both comprehension and the troubleshooting experience,
+a simplified conditional error message will be provided determined by
+Interactive or Script use. A cmdlet will provide the full error details when desired.
+
+## Motivation
+
+The on-screen experience, when receiving an error message,
+is controlled through the views NormalView (the default) and CategoryView. These are user selectable
+through the preference variable $ErrorView.
+This RFC describes changing the NormalView to a simplified conditional error message. The
+CategoryView should remain unchanged.
+A comprehensive detailed view of the fully qualified error, including inner exceptions,
+will be provided by the `Resolve-ErrorRecord` cmdlet.
+
+## Specification
+
+The proposal is to modify NormalView to a simplified conditional error message
+based on interactive versus script use.
+For in-depth troubleshooting, a new cmdlet `Resolve-ErrorRecord` to provide detailed error information.
+
+__Key Design Considerations__
+
+1. To reduce confusion and improve debugging success,
+error messages should call WriteErrorLine to produce a simplified message, including the word “ERROR:”
+to make consistent with Verbose and Warning messages. This is a change to the existing NormalView.
+
+- $ErrorView can refer to these views
+
+    + NormalView
+    + CategoryView
+
+- Simplified error message syntax from Interactive and script.
+
+![test](.\RFC00XX-Update-Error-View.png)
+
+2. A new cmdlet `Resolve-ErrorRecord` will produce comprehensive detailed
+view of the fully qualified error, including inner exceptions.
+
+- Resolve-ErrorRecord will provide the following:
+
+    + Display Last Error ($Error[0]) – default behavior
+    + Accept Pipeline input – support $error[1] |
+    + Option for the Newest X errors in the session
+
+- Resolve-ErrorRecord syntax
+
+```powershell
+Resolve-ErrorRecord  [-InputObject <psobject>] [-Newest <Int32>] [-All] [<CommonParameters>]
+```
+
+First parameter set
+
+- Newest
+
+    + Datatype: int32
+    + specifies one or more of last errors to display
+    + Not required
+
+__Example 1__
+Error occurs in Interactive mode. Cmdlet displays details of the last error displayed
+
+```powershell
+PS C:\> Get-Childitem -Path c:\notreal
+ERROR: Cannot find path ‘C:\notreal’ because it does not exist
+
+PS C:\test> Resolve-ErrorRecord
+
+**** Detailed message here ****
+```
+
+__Example 2__
+Error occurs in Script. Cmdlet displays error details from pipeline error object
+
+```powershell
+PS C:\> .\MyScript.ps1
+ERROR: Cannot find path 'C:\notreal' because it does not exist.
+  ---> C:\GitHub\pri-errorview\RustTest\test.ps1
+    |
+15  | Get-ChildItem -Path c:\notreal
+    |                     ^^^ Cannot find path 'C:\notreal' because it does not exist.
+    |
+    * Help: this is for additional help information
+
+PS C:\> $error[0] | Resolve-ErrorRecord
+
```

Comment:
There are comments about `Resolve-ErrorRecord` in this RFC that are being ignored. Additional comments have been added calling that out, and those comments plus the others they refer to are still being ignored. Why is that happening???

With respect to `Resolve-ErrorRecord`, how is what that would return different from a custom error format? It's just outputting a string, not an object, right? How about having an error view specific for that detailed error format? What about the suggestion for a `Get-Error` command instead that could be used for more than just error resolution, but also error processing, with a `Format-Error` command to allow users to see whatever format they want? This has been suggested elsewhere, but since that part of the discussion is being ignored, I feel I need to repeat it again here.